File: code/socket2/cxx/1_1_local_machine_info.cpp
#include "socket.hpp"
#include <iostream>

void print_machine_info() {
    // 1. 获取本机主机名
    std::string hostname = Socket::host::local_hostname();

    // 2. 解析主机名，拿到所有地址（IPv4 + IPv6）
    //    内部走 getaddrinfo，替代已废弃的 gethostbyname
    auto addrs = Socket::Address::resolve_all(hostname, 0);

    // 3. 输出
    std::cout << "Host name: " << hostname << "\n";
    for (const auto& addr : addrs) {
        std::cout << "IP address: " << addr.ip() << "\n";
    }
}

int main() {
    print_machine_info();
    return 0;
}

File: code/socket2/python/1_1_local_machine_info.py
#!/usr/bin/env python3
# Python Network Programming Cookbook -- Chapter - 1
# This program is optimized for Python 2.7.
# It may run on any other version with/without modifications.

import socket


def print_machine_info():
    host_name = socket.gethostname()
    ip_address = socket.gethostbyname(host_name)
    print("Host name: {}".format(host_name))
    print("IP address: {}".format(ip_address))

if __name__ == '__main__':
    print_machine_info()
